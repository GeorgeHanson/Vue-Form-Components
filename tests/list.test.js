import Vue from 'vue';
import ListComponent from '../src/Components/List.vue';

describe('ListComponent', () => {
    test("it generates the markup", () => {
        const renderer = require('vue-server-renderer').createRenderer();
        const vm = new Vue({
            el: document.createElement('div'),
            render: (createElement) => {
                return createElement(
                    ListComponent,
                    {
                        props: {
                            name: 'choices',
                            label: 'Please select a choice',
                            options: [1, 2, 3]
                        }
                    }
                )
            },
        });

        renderer.renderToString(vm, (err, str) => {
            expect(str).toMatchSnapshot();
        });
    });
});
