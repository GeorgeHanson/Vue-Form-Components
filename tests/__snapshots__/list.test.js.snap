// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListComponent it generates the markup 1`] = `
< div data - server - rendered = "true"
class = "form-group" > < div class = "col-sm-4" > < label class = "control-label" > Please select a choice: <!----></label> <!----></div> <div class="control-container col-sm-8"><div><!----> <!----> <select name="choices" id="choices" class="form-control"><!----> <option selected="selected"></option><option selected="selected"></option><option selected="selected"></option></select> <!----> <!----></div> <!----></div> <div class="clearfix"></div></div>
`;
